libs/community/langchain_community/vectorstores/pinecone.py

```diff
@@ -463,10 +456,16 @@ def from_existing_index(
         text_key: str = "text",
         namespace: Optional[str] = None,
         pool_threads: int = 4,
+        use_pod_based: Optional[bool] = True,
     ) -> Pinecone:
         """Load pinecone vectorstore from index name."""
-        pinecone_index = cls.get_pinecone_index(index_name, pool_threads)
-        return cls(pinecone_index, embedding, text_key, namespace)
+        pinecone = cls(
+            None, embedding, text_key, namespace, use_pod_based=use_pod_based
+        )
+        pinecone_index = pinecone.get_pinecone_index(index_name, pool_threads)
+        pinecone._index = pinecone_index
```

Comment:
don't love this, and comment above about keeping `get_pinecone_index` as a classmethod will likely address this.